src/main/nodejs/src/app/song-list/song-list.component.html

```diff
@@ -1,11 +1,12 @@
-<ul class="list-group shadow" *ngIf="songs">
-  <li *ngFor="let song of songs; trackBy: trackBySongCatalogId" (click)="gotoSong(song)" class="list-group-item list-group-item-action d-flex align-items-center no-gutters px-3 py-2">
+<ul class="list-group shadow" *ngIf="songs" cdkDropListGroup>
+  <li *ngFor="let song of songs; trackBy: trackBySongCatalogId; let i=index" cdkDropList (click)="gotoSong(song)" class="list-group-item list-group-item-action align-items-center no-gutters px-3 py-2" 
+    cdkDrag (mouseenter)="enter(i)" (cdkDropListDropped)="moveSong($event)" (cdkDragStarted)="previousIndex=i; dragging = true" (cdgDragDropped)="dragging = false">
```

Comment:
- Why remove `d-flex` and replace it with custom CSS?
- Dragging should NOT be allowed when this is not a playlist's songs (e.g. when it's a search result).